### PR TITLE
Refresh token implementation

### PR DESCRIPTION
**Pull request by TONYOG12**
access tokens refreshes expicitly after every request and stored in cookies. endpoints can also be restricted with jwt_requied() now

### CODE REVIEW

File: app/ents/employee/endpoints.py [hunk 1]
```diff
@@ -70,19 +72,18 @@ def login_employee():
...
 @bp.after_request
-# @jwt_required
 def refresh_expiring_jwts(success_response):
     try:
         employee_id = str(get_jwt_identity())
         if employee_id:
             exp_timestamp = get_jwt()["exp"]
             now = datetime.now(timezone.utc)
-            target_timestamp = datetime.timestamp(now + timedelta(minutes=30))
+            target_timestamp = datetime.timestamp(now + timedelta(minutes=2))
```
Comment (CorneliusOsei2) [inline app/ents/employee/endpoints.py:86]: Probably access the value from app.core.settings instead of hard-coding

File: app/main.py [hunk 2]
```diff
@@ -22,12 +24,17 @@ def register_blueprints(app: Flask) -> None:
     """Register `app` blueprints."""
     app.register_blueprint(employee_blueprint)
 
+def configure_jwt(app: Flask) -> None:
+    app.config['JWT_SECRET_KEY'] = settings.SECRET_KEY
```
Comment (CorneliusOsei2) [inline app/main.py:28]: I think there is a JWT_SECRET_KEY in app.core.settings / .env file. Let's use that instead. SECRET_KEY is for the whole api.

File: app/utilities/utils.py [hunk 3]
```diff
@@ -1,10 +1,9 @@
 from pydantic import ValidationError
-
+from flask import jsonify
```
Comment (CorneliusOsei2) [inline app/utilities/utils.py:2]: What is jsonify being used for?

File: app/core/security.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 from datetime import datetime, timedelta, timezone
 from http import HTTPStatus
+from flask import jsonify
```
Comment (CorneliusOsei2) [inline app/core/security.py:3]: What is jsonify being used for

File: app/core/security.py [hunk 5]
```diff
@@ -38,10 +41,10 @@ def create_token_with_id(self, userId):
         """Creates a JWT token with the `id` of the `user`."""
         token = jwt.encode(
             payload={
-                "subject": userId,
-                "expire_at": (datetime.utcnow() + timedelta(minutes=30)).ctime(),
+                "sub": userId,
+                "exp": (datetime.utcnow() + timedelta(minutes=2)).timestamp(),
             },
-            key=config["SECRET_KEY"],  # type: ignore
+            key=settings.SECRET_KEY,  # type: ignore
```
Comment (CorneliusOsei2) [inline app/core/security.py:47]: settings.JWT_SECRET_KEY may be better used instead of settings.SECRET_KEY

File: app/utilities/utils.py (hunk 3 above)
Comment (TONYOG12) [reply app/utilities/utils.py:2]: I removed the logic I was using that for I just forgot remove the import statement I'd push a commit to rectify that

File: app/core/security.py (hunk 5 above)
Comment (TONYOG12) [reply app/core/security.py:47]: noted ✅

File: app/ents/employee/endpoints.py (hunk 1 above)
Comment (TONYOG12) [reply app/ents/employee/endpoints.py:86]: alright. are we using 2 minutes expiration time tho?

Comment (CorneliusOsei2) [reply app/ents/employee/endpoints.py:86]: Yeah

File: app/utilities/utils.py (hunk 3 above)
Comment (CorneliusOsei2) [reply app/utilities/utils.py:2]: Sure sure

File: app/core/security.py [hunk 6]
```diff
@@ -38,10 +41,10 @@ def create_token_with_id(self, userId):
         """Creates a JWT token with the `id` of the `user`."""
         token = jwt.encode(
             payload={
-                "subject": userId,
-                "expire_at": (datetime.utcnow() + timedelta(minutes=30)).ctime(),
+                "sub": userId,
+                "exp": (datetime.utcnow() + timedelta(minutes=2)).timestamp(),
```
Comment (CorneliusOsei2) [inline app/core/security.py:45]: Could the 2 be replaced with settings.ACCESS_TOKEN_EXPIRE_MINUTES

File: app/ents/employee/endpoints.py [hunk 7]
```diff
@@ -70,19 +73,18 @@ def login_employee():
...
             token=security.create_token(employee),
         )
-
+    
     return error_response(error=EmployeeDoesNotExist.msg, code=HTTPStatus.BAD_REQUEST)
 
 
 @bp.after_request
-# @jwt_required
 def refresh_expiring_jwts(success_response):
```
Comment (CorneliusOsei2) [inline app/ents/employee/endpoints.py:81]: Can some of the logic here be extracted to app.core.security?

Comment (TONYOG12) [reply app/ents/employee/endpoints.py:81]: okay I will encaspulate that in there

Comment (TONYOG12) [reply app/ents/employee/endpoints.py:81]: kindly check latest commit to see if it looks good

File: app/core/security.py (hunk 6 above)
Comment (TONYOG12) [reply app/core/security.py]: oops must have missed that, latest commit resolves that

**Review by CorneliusOsei2 — APPROVED** [review]

**Pull request merged**